### PR TITLE
Fix scan unsupported workflow in SQl DB

### PR DESCRIPTION
**Pull request by yux0**
**What changed?**
1. Fix the result ordering of SQL
2. Fix version histories deserialization

**Why?**

**How did you test it?**
Local testing

**Potential risks**

### CODE REVIEW

**Review by yycptt — APPROVED** [review]

**Review by yycptt — COMMENTED** [review]
Can we add some comment for the new contract of list executions?

**Review by yycptt — COMMENTED** [review]
It looks like `minUUID` is also used by ListTaskList, will it cause the same issue there?

**Comment by yux0** [comment]
> Can we add some comment for the new contract of list executions?
Added comments about the behavior.

> It looks like `minUUID` is also used by ListTaskList, will it cause the same issue there?

I think so. Added a comment on this API. This API is only used in Scanner.

File: common/persistence/sql/sqlTaskManager.go [hunk 1]
```diff
@@ -264,6 +264,9 @@ type taskListPageToken struct {
 	TaskType int64
 }
 
+// ListTaskList lists tasklist from DB
+// DomainID translates into byte array in SQL. The minUUID is not the minimum byte array.
+// This API could return incomplete result set.
```
Comment (yycptt) [inline common/persistence/sql/sqlTaskManager.go:269]: Shall we create an issue for this? Looks like this is something we need to fix.

Comment (yux0) [reply common/persistence/sql/sqlTaskManager.go:269]: https://github.com/uber/cadence/issues/3911

**Pull request merged**